assets/router/deployment.yaml

```diff
@@ -15,11 +15,22 @@ spec:
     metadata:
       labels:
         app: router
+        ingress.openshift.io/component: ingress-controller
     spec:
       serviceAccountName: router
       nodeSelector:
         beta.kubernetes.io/os: linux
         node-role.kubernetes.io/worker: ''
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - topologyKey: "kubernetes.io/hostname"
+            labelSelector:
+              matchExpressions:
+              - key: ingress.openshift.io/component
+                operator: In
+                values:
+                - ingress-controller
```

Comment:
I do not see any issue with the code but do we need to be concerned about performance issues on large clusters?

https://kubernetes.io/docs/concepts/configuration/assign-pod-node/  has this note:
```
Note: Inter-pod affinity and anti-affinity require substantial amount of processing which can slow down scheduling in large clusters significantly. We do not recommend using them in clusters larger than several hundred nodes.
```

kube-dns adopted and later reverted pod anti-affinity rules:
https://github.com/kubernetes/kubernetes/pull/54166
https://github.com/kubernetes/kubeadm/issues/995

I'm not sure how many nodes do we promise to support as part of OpenShift 4.x installation. Do you think it is safer to add a knob to turn off pod anti-affinity if needed for ingress controller?

Reply:
@derekwaynecarr @smarterclayton @pravisankar haven't dug too deeply yet, but looks like the problems currently start at scales at or approaching 5k nodes?

https://github.com/kubernetes/kubernetes/issues/54189

Reply:
Decided given the focused optimizations for this particular AntiAffinity rule we're okay with the possible scale implications.